### ISSUE TITLE
emitter/router doesn't work immediately

### ISSUE THREAD

**Issue by blahah** [yoshuawuyts/choo#470]
in v5, if I have a function like so...

```js
app.use((state, emitter) => {
  emitter.emit('pushState', '/someroute')
})
```

it doesn't do anything.

Whereas if I do:

```js
app.use((state, emitter) => {
  setTimeout(emitter.emit('pushState', '/someroute'), 200)
})
```

it works.

It feels as though the `app.use()` callbacks shouldn't be called until the api is ready to be used.

**Comment by yoshuawuyts** [comment]
ya you should probably wait for `emitter.on('DOMContentLoaded')` before
firing these events like as a general rule - that way they always fire
after start() has taken off

**Comment by blahah** [comment]
@yoshuawuyts is there any reason why the `.use()` functions should be loaded before then? If the app just waited for `DOMContentLoaded` before calling the callbacks, then the user doesn't have to worry about waiting for it explicitly.

If \o isn't going to happen, the need to wait for the event should definitely be documented.

**Comment by yoshuawuyts** [comment]
.use() functions should be immediate so default data can be setup; views
would otherwise become littered with checks if properties exist. You're
right, we should document this better - PR would be very welcome here!

**Comment by yoshuawuyts** [comment]
Closed in #472

**Comment by Tyler-Murphy** [comment]
Could `pushState` and `replaceState` events be queued by choo until `documentReady`?

It's nice not to think about when the DOM is ready. Choo already hides this detail for us in [`mount`](https://github.com/yoshuawuyts/choo/blob/8ff7deea3cc0b1041338068211b3cf5662df8b2e/index.js#L109).
